### PR TITLE
Favorites icon + side menu to graphs being displayed

### PR DESCRIPTION
**Pull request by lovedeep92**
**JIRA:**
See ticket here - https://jira.rax.io/browse/MNRVA-821

**Description:**

1) Add favourite icon to each graph.
2) Add favourite tab to left nav.

**Testing:**

`npm run test:unit`

**Screenshot:**

Screenshot 2021-07-23 at 8 31 12 PM

(if applicable)

### CODE REVIEW

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.html [hunk 1]
```diff
@@ -29,6 +29,14 @@
       </div>
 
     </div>
+
```
Comment (pras7430) [inline src/app/_features/visualize/pages/visualize.page/visualize.page.component.html:32]: we can provide an id

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.scss [hunk 2]
```diff
@@ -42,4 +42,19 @@
...
 .center{
     text-align: center;  
+}
+.favorite {
+    background: transparent;
+    border: none;
+    cursor: pointer;
+    background:#d9d9d9;
+}
+.img{
```
Comment (pras7430) [inline src/app/_features/visualize/pages/visualize.page/visualize.page.component.scss:52]: classes name are not relevent

File: src/app/_shared/components/hedwig-graph/hedwig-graph.component.html [hunk 3]
```diff
@@ -1,11 +1,23 @@
 
 <div class="hxRow">
-    <div *ngIf="data"  class="hxCol hxSpan-6 metricNamendGrp leftAlign">
+  
+  <div *ngIf="data"  class="hxCol hxSpan-6 metricNamendGrp leftAlign">
+    <div class="hxRow">
+      <div class="hxCol hxSpan-2">
+        <button type="button" class="faStar" (click)="setFav()">
+          <img *ngIf="!favStatus" src="https://img.icons8.com/ios-glyphs/30/000000/star--v2.png"/>
```
Comment (pras7430) [inline src/app/_shared/components/hedwig-graph/hedwig-graph.component.html:8]: I am not sure about it.. but we should use any of the Helix icons or we can ask the helix team to add this icon
@FarajiA

File: src/app/_shared/components/hedwig-graph/hedwig-graph.component.ts [hunk 4]
```diff
@@ -31,4 +32,7 @@ export class HedwigGraphComponent implements OnInit {
   ngOnInit(): void {
  
   }
+  setFav() {
+      this.favStatus = this.favStatus ? false : true;
```
Comment (pras7430) [inline src/app/_shared/components/hedwig-graph/hedwig-graph.component.ts:36]: we can make it more simple 

this.favStatus = !this.favStatus

File: src/app/_shared/components/hedwig-graph/hedwig-graph.component.html (hunk 3 above)
Comment (FarajiA) [reply src/app/_shared/components/hedwig-graph/hedwig-graph.component.html:8]: I'm okay with using an icon that's not in the Helix library. A few things:
- We cannot link to across the internet this request may fail in staging/prod and is generally bad practice you will have to get the image and reference it locally
- An SVG would be preferred - you will need to have a selected stated & an un-selected state to represent both

Comment (lovedeep92) [reply src/app/_shared/components/hedwig-graph/hedwig-graph.component.html:8]: got it @FarajiA , I will change this!

File: src/app/_shared/components/hedwig-graph/hedwig-graph.component.ts (hunk 4 above)
Comment (lovedeep92) [reply src/app/_shared/components/hedwig-graph/hedwig-graph.component.ts:36]: done!

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.scss (hunk 2 above)
Comment (lovedeep92) [reply src/app/_features/visualize/pages/visualize.page/visualize.page.component.scss:52]: changed the name!

Comment (lovedeep92) [reply src/app/_features/visualize/pages/visualize.page/visualize.page.component.scss:52]: deleted!

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.html (hunk 1 above)
Comment (lovedeep92) [reply src/app/_features/visualize/pages/visualize.page/visualize.page.component.html:32]: not getting it? which id?

File: src/app/_shared/components/hedwig-graph/hedwig-graph.component.ts [hunk 5]
```diff
@@ -16,6 +16,7 @@ type="MAAS_cpu" [data]="data"></app-hedwig-graph>
 export class HedwigGraphComponent implements OnInit {
 
   JSON: JSON = JSON;
+  favStatus:boolean = false;
```
Comment (pras7430) [inline src/app/_shared/components/hedwig-graph/hedwig-graph.component.ts:19]: favStatus = false

**Review by pras7430 — APPROVED** [review]

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.scss [hunk 6]
```diff
@@ -42,4 +42,16 @@
   }
 .center{
     text-align: center;  
+}
+.favIcon{
+  width: 3%;
+}
+.imgDiv{
```
[the remark this answers is not in the document]
Comment (lovedeep92) [reply src/app/_features/visualize/pages/visualize.page/visualize.page.component.scss:49]: not used anymore

File: src/app/_shared/components/hedwig-graph/hedwig-graph.component.ts (hunk 5 above)
Comment (lovedeep92) [reply src/app/_shared/components/hedwig-graph/hedwig-graph.component.ts:19]: okay !

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.scss [hunk 7]
```diff
@@ -42,4 +42,16 @@
   }
 .center{
     text-align: center;  
+}
+.favIcon{
```
[the remark this answers is not in the document]
Comment (lovedeep92) [reply src/app/_features/visualize/pages/visualize.page/visualize.page.component.scss:46]: changed!

File: src/app/_shared/components/hedwig-graph/hedwig-graph.component.html [hunk 8]
```diff
@@ -1,9 +1,20 @@
 
 <div class="hxRow">
-    <div *ngIf="data"  class="hxCol hxSpan-6 metricNamendGrp leftAlign">
+  
+  <div *ngIf="data"  class="hxCol hxSpan-6 metricNamendGrp leftAlign">
+    <div class="hxRow">
+      <div class="hxCol hxSpan-2">
+        <button type="button" class="faStar" (click)="setFav()">
+          <img *ngIf="!favStatus" src="./../../../../assets/un-selected-star.png"/>
+          <img *ngIf="favStatus" src="./../../../../assets/selected-star.png"/>
```
Comment (FarajiA) [inline src/app/_shared/components/hedwig-graph/hedwig-graph.component.html:9]: So you really want to bring both images into the template you can switch between the two with classes that load them

Comment (lovedeep92) [reply src/app/_shared/components/hedwig-graph/hedwig-graph.component.html]: changed and added SVG!

**Pull request merged**